django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js

```diff
@@ -87,13 +87,41 @@
         }
     }
 
+    function updateRelatedSelectsOptions(currentSelect, win, objId, newRepr, newId) {
+        // After create/edit a Model from the options next to the current SELECT (+ or :pencil:)
+        // update ForeignKey PK of the rest of SELECTs in the page
+
+        const path = win.location.pathname;
+        // we have to extract the model from the popup url
+        // '.../<model>/add/' or '.../<model>/<id>/change/'
+        // depending the action (ADD or CHANGE)
+        const modelName = path.split('/')[path.split('/').length - (objId ? 4 : 3)];
+        const selectsRelated = document.querySelectorAll(`[data-model-ref="${modelName}"] select`);
+
+        selectsRelated.forEach(function(select) {
+            if (currentSelect !== select) {
+                if (!objId) {
+                    select.options[select.options.length] = new Option(newRepr, newId);
```

Comment:
Add return statement to remove else statement below.

Reply:
I added the `return` statement, removed the `else` and also updated the code to avoid looping through all options.

Thank you!